Realm.podspec

```diff
@@ -1,89 +1,136 @@
+  # Platforms
+  s.ios.deployment_target     = '7.0'
+  s.osx.deployment_target     = '10.9'
+  s.tvos.deployment_target    = '9.0'
+  s.watchos.deployment_target = '2.0'
+  
+  # Compilation
+  if ENV['REALM_DISABLE_SYNC']
+    s.dependency             'RealmCore'
+    s.libraries            = 'c++'
+    s.header_mappings_dir  = 'Realm'
+    s.source_files         = 'Realm/*.{h,hpp,m,mm}',
+                             'Realm/ObjectStore/src/*.{h,hpp,cpp}',
+                             'Realm/ObjectStore/src/impl/*.{h,hpp,cpp}',
+                             'Realm/ObjectStore/src/impl/apple/*.{h,hpp,cpp}',
+                             'Realm/ObjectStore/src/util/*.{h,hpp,cpp}',
+                             'Realm/ObjectStore/src/util/apple/*.{h,hpp,cpp}'
+    s.public_header_files  = 'Realm/Realm.h',
+                             'Realm/RLMArray.h',
+                             'Realm/RLMCollection.h',
+                             'Realm/RLMConstants.h',
+                             'Realm/RLMListBase.h',
+                             'Realm/RLMMigration.h',
+                             'Realm/RLMObject.h',
+                             'Realm/RLMObjectBase.h',
+                             'Realm/RLMObjectSchema.h',
+                             'Realm/RLMOptionalBase.h',
+                             'Realm/RLMPlatform.h',
+                             'Realm/RLMProperty.h',
+                             'Realm/RLMRealm.h',
+                             'Realm/RLMRealmConfiguration.h',
+                             'Realm/RLMResults.h',
+                             'Realm/RLMSchema.h',
+                             
+                             # Dynamic
+                             'Realm/*_Dynamic.h',
+    s.private_header_files = 'Realm/{*_Private,RLMAccessor,RLMObjectStore}.h'
+    s.exclude_files        = 'Realm/Tests', 'Realm/ObjectStore/**/sync*',
+                             'Realm/RLMSync*',
+                             'Realm/RLMTokenModels.{h,m}',
+                             'Realm/RLMRealmConfiguration+Sync.{h,mm}',
+                             'Realm/RLMNetworkClient.{h,m}',
+                             'Realm/RLMAuthResponseModel.{h,m}',
+    s.module_map           = 'Realm/module.modulemap'
+    s.prepare_command      = 'touch Realm/RLMPlatform.h'
+    s.compiler_flags       = '-D__ASSERTMACROS__',
+                             "-DREALM_COCOA_VERSION='@\"#{s.version}\"'",
+                             '-DREALM_ENABLE_ASSERTIONS',
+                             '-DREALM_ENABLE_ENCRYPTION'
+    s.pod_target_xcconfig  = { 'APPLICATION_EXTENSION_API_ONLY' => 'YES',
+                               'CLANG_CXX_LANGUAGE_STANDARD' => 'c++14',
+                               'HEADER_SEARCH_PATHS' => '"${PODS_ROOT}/RealmCore/src" "${PODS_ROOT}/Realm/Realm/ObjectStore/src"' }
+  else
+    s.libraries            = 'c++', 'z'
+    public_header_files    = 'include/**/RLMArray.h',
```

Comment:
If we do go this route I'd rather have the non-sync files added unconditionally and have this just add more files rather than overriding the entire list.